### ISSUE TITLE
Bug Report: ButtonGroup overflow popup menu moves on scroll and is not positioned correctly.

### ISSUE THREAD

**Issue by bensontrent** [stripe/stripe-apps#945]
When a ButtonGroup is in overflow mode, the popup menu should be absolutely-positioned to the right, relative of the button group.  

![image](https://github.com/stripe/stripe-apps/assets/5982592/7cb470a6-3a7c-4bed-be39-9411f241146d)

You can see the problem in the animation:  The popup menu is cut off by the end of the screen and when a user scrolls their window, the pop up menu follows the scroll, not the button group:
 
![popup-menu](https://github.com/stripe/stripe-apps/assets/5982592/59a45140-ad42-4fd7-8433-6b6a2cfafd75)

(The framerate of my animated gif is not sufficient to show the scrolling motion but the menu scrolls off-screen on scroll)

```
<ButtonGroup>
    <Button css={{ width: "fill" }} onPress={() => { saveShipmentAndNavigate(`/shipping-options/${id}`) }}>
         Edit options</Button>
    <Button type="primary" css={{ width: "fill" }} disabled={isBuying || context.environment.mode === "test"} 
         onPress={() => handleBuyOrTest("production")} > Buy label</Button>
    <Button type="secondary" css={{ alignX: "end", }} disabled={isBuying} 
        onPress={() => handleBuyOrTest("test")} > Print test label</Button>
</ButtonGroup>
```

**Comment by gabrielhurley-stripe** [comment]
Yep, that doesn't look right! I've flagged this internally and we'll check it out. Thanks!

**Comment by bensontrent** [comment]
I do hope your team will be at Stripe Sessions in April in SF.  I'll be there!  Having been part of a 5000+ team of software developers I know your pain and I hope I can be a small part of your developer  success story! If the Stripe Management Team were smart, they would be sending priority resources to your team! I only hope for you success! I hope my shipping app, when made pubic, will only add value to the of having Stripe as part their company of the tech stack.
